Sources/Sentry/SentryDispatchQueueWrapper.m

```diff
@@ -56,23 +56,41 @@ - (void)dispatchSyncOnMainQueue:(void (^)(void))block
     }
 }
 
+- (id)dispatchSyncOnMainQueueWithResult:(id (^)(void))block {
+	return [self dispatchSyncOnMainQueueWithResult:block timeout:DISPATCH_TIME_FOREVER];
+}
+
 - (BOOL)dispatchSyncOnMainQueue:(void (^)(void))block timeout:(NSTimeInterval)timeout
 {
-    if ([NSThread isMainThread]) {
-        block();
-    } else {
-        dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
-
-        dispatch_async(dispatch_get_main_queue(), ^{
-            block();
-            dispatch_semaphore_signal(semaphore);
-        });
+	NSNumber *result = @0;
+	result = [self dispatchSyncOnMainQueueWithResult:^id _Nonnull{
+		block();
+		return @1;
+	} timeout:timeout];
+	return result.boolValue;
+}
 
-        dispatch_time_t timeout_t
-            = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(timeout * NSEC_PER_SEC));
-        return dispatch_semaphore_wait(semaphore, timeout_t) == 0;
-    }
-    return YES;
+- (id)dispatchSyncOnMainQueueWithResult:(id (^)(void))block timeout:(NSTimeInterval)timeout
+{
+	if ([NSThread isMainThread]) {
```

Comment:
nit: worth using `DispatchSpecificKey` for the main queue instead.

Reply:
Is there any benefit using `DispatchSpecificKey` instead of `[NSThread isMainThread]`?
I think the goal here is to ensure this is running on main thread. Setting a specificKey to main dispatch-queue and check the current dispatch-queue's specificKey may be excrescent.